core/src/main/scala/cats/Selective.scala

```diff
@@ -27,12 +28,46 @@ trait Selective[F[_]] {
     branch(condition)(left)(right)
   }
 
-  // TODO more combinators here
+  def whenS[A](fbool: F[Boolean])(fa: F[Unit]): F[Unit] =
+    ifS(fbool)(Functor[F].as(fa, ()))(().pure[F])
+
+  def bindBool[A](fbool: F[Boolean])(f: Boolean => F[A]): F[A] =
+    ifS(fbool)(f(false))(f(true))
+
+  def fromMaybeS[A](fa: F[A])(fm: F[Option[A]]): F[A] =
+    select(
+      Functor[F].map(fm)(
+        _.fold[Either[Unit, A]](
+          Left.apply(())
+        )(Right.apply)
+      )
+    )(Functor[F].map(fa)(a => Function.const(a)))
+
+  def orS(fbool: F[Boolean])(fa: F[Boolean]): F[Boolean] =
+    ifS(fbool)(true.pure[F])(fa)
 
+  def andS(fbool: F[Boolean])(fa: F[Boolean]): F[Boolean] =
+    ifS(fbool)(fa)(false.pure[F])
+
+  def anyS[G[_]: Foldable, A](test: A => F[Boolean])(ga: G[A]): Eval[F[Boolean]] =
+    Foldable[G].foldRight(ga, Eval.later(false.pure[F]))({
```

Comment:
Isn't `Eval.now` more suitable here, as it's a pure value? Saves the allocation of a thunk.